developing/how-to-query-nova-db-soft-deleted-instances.md

```diff
@@ -0,0 +1,70 @@
+# How to query nova db for soft deleted instances
+
+This is useful for verifying or debugging archive and purge steps.
```

Comment:
(non-blocking) Following from @gabrielcocenza 's comment, it might be good to put this line to `docs/ how-to archive-old-data.rst ` and put a link to this doc `developing/how-to-query-nova-db-soft-deleted-instances.md` if the user is interested in verifying the archive and purge

Reply:
Agreed, it could be better elsewhere, but I'm not sure where is best.  We can refactor later I guess; at least we have it recorded somewhere for now. :)